src/Illuminate/Routing/Exceptions/UrlGenerationException.php

```diff
@@ -3,17 +3,35 @@
 namespace Illuminate\Routing\Exceptions;
 
 use Exception;
+use Illuminate\Routing\Route;
+use Illuminate\Support\Str;
 
 class UrlGenerationException extends Exception
 {
     /**
      * Create a new exception for missing route parameters.
      *
      * @param  \Illuminate\Routing\Route  $route
+     * @param array $parameters
      * @return static
      */
-    public static function forMissingParameters($route)
+    public static function forMissingParameters(Route $route, array $parameters = [])
```

Comment:
We'll have to be aware that this is technically a breaking change for people extending this exception and method.

Reply:
We could just re-target this PR at master. It's not long till another release anyway (indeed, it never really is, with the 6 month release cycle).

Reply:
I'm happy with whatever. :)